packages/perseus/src/widgets/interactive-graphs/graphs/quadratic.tsx

```diff
@@ -0,0 +1,120 @@
+import {color} from "@khanacademy/wonder-blocks-tokens";
+import {Plot} from "mafs";
+import * as React from "react";
+
+import {movePoint} from "../reducer/interactive-graph-action";
+
+import {StyledMovablePoint} from "./components/movable-point";
+
+import type {QuadraticGraphState, MafsGraphProps} from "../types";
+import type {vec} from "mafs";
+
+type QuadraticGraphProps = MafsGraphProps<QuadraticGraphState>;
+type QuadraticCoords = QuadraticGraphState["coords"];
+type QuadraticCoefficient = [number, number, number];
+
+export function QuadraticGraph(props: QuadraticGraphProps) {
+    const {dispatch, graphState} = props;
+
+    const {coords, range} = graphState;
+
+    // The coefficients are used to calculate the quadratic equation, plot the graph, and to indicate
+    // to content creators the currently selected "correct answer". ex: y = 0.200x^2 + 0.000x + 0.000
+    // While we should technically never have invalid coordinates, we want to ensure that we have a fallback
+    const coeffRef = React.useRef<QuadraticCoefficient>([0, 0, 0]);
+    const coeffs = getQuadraticCoefficients(coords);
+    if (coeffs !== undefined) {
+        coeffRef.current = coeffs;
+    }
+
+    // Destructure the coefficients for calculating the quadratic equation
+    const [a, b, c] = coeffRef.current;
+
+    // Calculate the y value based on the current x value
+    const y = (x) => (a * x + b) * x + c;
+
+    // Set the xy values for the Parametric plot
+    const xy = (x: number) => [x, y(x)] as vec.Vector2;
+
+    // Determine the range / domain of the graph
+    const t: vec.Vector2 = [range[0][0], range[0][1]] as vec.Vector2;
+
+    const handleOnMove = (destination: vec.Vector2, elementId: number) => {
+        // If the destination is invalid, we want do not want to move the point
+        const validDestination = isValidDestination(
+            destination,
+            elementId,
+            coords,
+        );
+        if (validDestination === false) {
+            return;
+        }
+
+        dispatch(movePoint(elementId, destination));
+    };
+
+    return (
+        <>
+            <Plot.Parametric xy={xy} t={t} color={color.blue} />
```

Comment:
Is `Plot.Parametric` needed here? Mafs also has a `Plot.OfX` component that just lets you pass a function to compute y from x. https://mafs.dev/guides/display/plots